Sources/Subscription/AccountManager.swift

```diff
@@ -221,19 +230,44 @@ public class AccountManager: AccountManaging {
         }
     }
 
-    public func fetchEntitlements() async -> Result<[Entitlement], Error> {
-        guard let accessToken else { return .failure(EntitlementsError.noAccessToken) }
+    private func fetchRemoteEntitlements() async -> Result<[Entitlement], Error> {
+        guard let accessToken else {
+            subscriptionCache.cleanup()
+            return .failure(EntitlementsError.noAccessToken)
+        }
+
+        let cachedEntitlements: [Entitlement]? = subscriptionCache.object(forKey: Constants.cachedEntitlementsKey)
 
         switch await AuthService.validateToken(accessToken: accessToken) {
-        case .success(let response):
+
+            case .success(let response):
             let entitlements = response.account.entitlements.compactMap { Entitlement(rawValue: $0.product) }
-            return .success(entitlements)
+                if entitlements != cachedEntitlements {
+                    subscriptionCache.set(entitlements, forKey: SubscriptionCache.Component.entitlements.rawValue)
+                    NotificationCenter.default.post(name: .entitlementsUpdated, object: self, userInfo: nil)
+                }
+                return .success(entitlements)
+
+            case .failure(let error):
+                subscriptionCache.cleanup()
+                os_log(.error, log: .subscription, "[AccountManager] fetchEntitlements error: %{public}@", error.localizedDescription)
+                return .failure(error)
+            }
+        }
+
+        public func fetchEntitlements(skipCache: Bool = false) async -> Result<[Entitlement], Error> {
```

Comment:
Maybe I’m overthinking this, but what if instead of `skipCache` we do something similar to `NSURLRequest.CachePolicy`? E.g. supporting these scenarios based on what the API user wants:

- `reloadIgnoringLocalCacheData`: Same as `skipCache`
- `returnCacheDataElseLoad`: Use existing cache, load from remote only if there’s no cached data
- `returnCacheDataDontLoad`: Use existing cache, fails if no cached data is available

I can see the entitlement monitoring code that NetP has uses `returnCacheDataElseLoad` behavior.